src/Meziantou.Analyzer/Rules/OptimizeLinqUsageAnalyzer.cs

```diff
@@ -564,6 +582,53 @@ bool HasTake()
             }
         }
 
+        private static void UseCastInsteadOfSelect(OperationAnalysisContext context, IInvocationOperation operation)
+        {
+            if (!string.Equals(operation.TargetMethod.Name, nameof(Enumerable.Select), StringComparison.Ordinal))
+                return;
+
+            // A valid 'Select' operation always has 2 arguments, regardless of whether the underlying code syntax is
+            // that of a call to an extension method:   source.Select(selector);
+            // or to a static method:                   Enumerable.Select(source, selector);
+            //  Operation's first argument  -> 'source'
+            //  Operation's second argument -> 'selector'
+            if (operation.Arguments.Length != 2)
+                return;
+
+            var selectorArg = operation.Arguments[1];
+
+            // Get the selector's return operation
+            var returnOp = selectorArg.Descendants().OfType<IReturnOperation>().FirstOrDefault();
+            if (returnOp is null)
+                return;
+
+            // Is it returning a cast value?
+            if (!(returnOp.ReturnedValue is IConversionOperation castOp))
+                return;
+
+            // And is the cast applied to the sequence element, passed in as argument to the selector?
+            if (!(castOp.Operand is IParameterReferenceOperation))
+                return;
+
+            if (!(castOp.Syntax is CastExpressionSyntax castExpression))
+                return;
+
+            // Get the location of the [|Select|] member access expression
+            if (!(operation.Syntax.ChildNodes().FirstOrDefault() is MemberAccessExpressionSyntax memberAccessExpression))
+                return;
+
+            var castType = castExpression.Type.ToString();
```

Comment:
I think this is equivalent to `castOp.Type`.

Reply:
I opted for
 `castOp.Type.ToMinimalDisplayString(operation.SemanticModel, operation.Syntax.SpanStart);`
instead.

This gives me, for instance, `object` rather than `Object`.